### DISCUSSION TITLE
Advanced User Onbaording: Match made in OSS

### DISCUSSION THREAD

**Discussion by compgeniuses** [DefGuard/defguard#684] [Ideas]
I noticed that Defguard has a focus on the initial user onboarding experience.

I recently came across an onboarding solution, that also has OIDC support called [Chiefonbaording](https://chiefonboarding.com/)
And Upon Digging in it, there is also an intergation it has (still work in progress) that Allows users account provision into [Authentik](https://integrations.chiefonboarding.com/manifest/10)

Equally,since the solution Allows for login into it using [OIDC](https://docs.chiefonboarding.com/config/oidc.html) and Defguard is aiming to put a focus on the User onbaording proccess. it got me thinking what a potential Integration would Look like.

Implementing and Allowing Chiefonbaording to PRovision USers into Defguard, and Allowing those Users to login into Chiefonbaording using Defguard provided OIDC.

With the Implementation of [Allowing external OIDC users in Defguard](https://github.com/DefGuard/defguard/issues/602) the possibility become endless. on their repo, there is a [PR for implementing OIDC using AllAuth(pythin)](https://github.com/chiefonboarding/ChiefOnboarding/pull/387) which might offer these kinds of integrations easier. According to [this](https://github.com/chiefonboarding/ChiefOnboarding/issues/437#issuecomment-2027464986) the PR could be prioritized with a 2k Donation

**Comment by kchudy** [comment]
@compgeniuses Please elaborate on the workflow you're proposing.

If I understood you correctly, you want to have Chiefonbaording configured with Defguard as an identity provider, so users sign in to Chiefonbaording using Defguard.

You also want to integrate Chiefonbaording with Defguard (possibly using Defguard API), so whenever a user is created in Chiefonbaording, they're also created in Defguard, allowing the user to sign in to their profile in Chiefonbaording using Defguard.

Is that what you're trying to achieve?

**Reply by compgeniuses** [reply]
Chiefonboarding is an onboarding tool for organizations, that allows new employees to be taken through the entire user journey of being onboarded into an organization.

 user is created on Chiefonboarding Either Manually, or through triggers, then a serioes of steps are taken through IT admin, to create the new user, their AD accounts, Potentially SSO accounts, to be able to login to other services etc.
 
 Since Defguard, offers a partially user self onboarding process, 
 
 Defguard, will be Configured as SSO provider for Chiefonboarding, then a new Employee added into Chiefonboarding, will trigger, Defguard, to create the New User. Which will follow, the onboarding of the new user, through the Defguard onboarding processes, which will follow up with additional tasks within Chiefonboarding, that the user, now has to login, via SSO, using Defguard, and in the company, they continue accessing other services and Utilities.

**Reply by kchudy** [reply]
We're not planning that kind of integration in the near future, but we'll keep that in mind. With Defguard's API, it should be possible to trigger user creation and enrollment from Chiefonboarding.
